### PR TITLE
[PART 2] pgx v5 support (#1823)

### PR DESCRIPTION
**Pull request by mcdoker18**
Added support for v5 of the pgx Go drives

* 'pgx/v5' option requires '--experimental' cli flag.
* array, ranges, multiranges uses generic version for 'pgx/v5'. It supports multidemensional arrays.
* 'pgx/v5' implementation prefers use pgtype types. For example, pgtype.Bool instead of sql.NullBool.
* added support for new pg types:
  * datemultirange
  * tsmultirange
  * tstzmultirange
  * nummultirange
  * int4multirange
  * int8multirange
  * bit
  * varbit
  * cid
  * oid
  * tid
  * circle
  * line
  * lseg
  * path
  * point
  * polygon

### CODE REVIEW

File: internal/codegen/golang/postgresql_type.go [hunk 1]
```diff
@@ -122,6 +151,8 @@ func postgresType(req *plugin.CodeGenRequest, col *plugin.Column) string {
 
 	case "jsonb":
 		switch driver {
+		case SQLDriverPGXV5:
+			return "[]byte"
```
[the remark this answers is not in the document]
Comment (mcdoker18) [reply internal/codegen/golang/postgresql_type.go:155]: BTW, user can override default []byte type with some struct.

For example:

```golang
type SomeDTOStruct struct {
	Value1 int       `json:"value_1"`
	Value2 string `json:"value_2"`
}
```

```json
...
      "overrides": [
        {
          "column": "some_table.jsonb_field",
          "go_type": "*example.com/db/dto.SomeDTOStruct"
        }
      ]
...
```

Then pgx v5 will marshall/unmarshall JSON to struct itself.

I can add it to the docs, if it's not obvious.

File: internal/codegen/golang/postgresql_type.go [hunk 2]
```diff
@@ -179,16 +245,21 @@ func postgresType(req *plugin.CodeGenRequest, col *plugin.Column) string {
...
 	case "cidr":
 		switch driver {
+		case SQLDriverPGXV5:
+			if notNull {
+				return "netip.Prefix"
+			}
+			return "*netip.Prefix"
 		case SQLDriverPGXV4:
-			return "pgtype.CIDR"
-		case SQLDriverLibPQ:
```
Comment (kyleconroy) [inline internal/codegen/golang/postgresql_type.go:184]: Removing this cases the build to break. Can we add it back in?

Comment (mcdoker18) [reply internal/codegen/golang/postgresql_type.go]: Fixed

**Comment by JordanP** [comment]
I am interesting in this PR too. Anything I could do to help ?

**Comment by mcdoker18** [comment]
Hi @kyleconroy Can you please take a look again? I fixed all merge conflicts

**Review by kyleconroy — APPROVED** [review]

File: internal/codegen/golang/postgresql_type.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (f1forhelp) [reply internal/codegen/golang/postgresql_type.go:186]: Seems like above mentioned syntax for absolute path ("go_type": "*example.com/db/dto.SomeDTOStruct") not working . Throwing error path not recognised in latest go version.

[the remark this answers is not in the document]
Comment (mcdoker18) [reply internal/codegen/golang/postgresql_type.go:186]: @f1forhelp if you are sure that there is a bug in the sqlc you can create an issue with reproducible example.

[the remark this answers is not in the document]
Comment (f1forhelp) [reply internal/codegen/golang/postgresql_type.go:186]: @mcdoker18  its working.

I used these json values.

          "overrides": [
            {
              "column": "events.event_price",
              "go_type": {
                "import":"test_proj/models/sqlc_json_model",
                "package": "sqlcjsonmodel",
                "type":"EventPrice"
              }
            }
          ],

**Pull request merged**